internal/xds/translator/session_persistence.go

```diff
@@ -61,28 +60,41 @@ func (s *sessionPersistence) patchHCM(mgr *hcmv3.HttpConnectionManager, irListen
 			continue
 		}
 
-		var sessionCfg proto.Message
-		var configName string
+		var (
+			sessionCfgAny *anypb.Any
+			err           error
+			configName    string
+		)
 		switch {
 		case sp.Cookie != nil:
 			configName = cookieConfigName
-			sessionCfg = &cookiev3.CookieBasedSessionState{
+			sessionCfg := &cookiev3.CookieBasedSessionState{
 				Cookie: &httpv3.Cookie{
 					Name: sp.Cookie.Name,
 					Path: routePathToCookiePath(route.PathMatch),
-					Ttl:  durationpb.New(sp.Cookie.TTL.Duration),
 				},
 			}
+
+			if sp.Cookie.TTL != nil {
+				sessionCfg.Cookie.Ttl = durationpb.New(sp.Cookie.TTL.Duration)
+			}
+
+			sessionCfgAny, err = anypb.New(sessionCfg)
+			if err != nil {
+				return fmt.Errorf("failed to marshal %s config: %w", egv1a1.EnvoyFilterSessionPersistence.String(), err)
+			}
+
 		case sp.Header != nil:
 			configName = headerConfigName
-			sessionCfg = &headerv3.HeaderBasedSessionState{
+			sessionCfg := &headerv3.HeaderBasedSessionState{
 				Name: sp.Header.Name,
 			}
-		}
 
-		sessionCfgAny, err := anypb.New(sessionCfg)
-		if err != nil {
-			return fmt.Errorf("failed to marshal %s config: %w", egv1a1.EnvoyFilterSessionPersistence.String(), err)
```

Comment:
Any reason you moved this `sessionCfgAny, err := anypb.New(sessionCfg)` to each `case` of the above `switch`? Looks like just making a duplicated code.

Reply:
ack, address this, ptal